community_mailbot/scripts/forward_discourse.py

```diff
@@ -0,0 +1,203 @@
+# encoding: utf-8
+"""
+Forward new topics in selected Discourse categories to select email addresses
+using Mandrill.
+
+This script is intended to be run as a repeated task that pulls new topics
+and forward the HTML content of the topic's first post to email recipents
+through Mandrill. When topics are forwarded, they are also cached.
+
+Specific Discourse categories can be monitoried, and each category can have
+topics forwarded to one or more email recipients. The map file (see below)
+figures the email recipeints for selected categories.
+
+Map file
+--------
+
+The ``map_file`` argument should be set to the path of a JSON file that
+category IDs to email recipients. The format is::
+
+    {
+        "<id>": [{"email": "<email address>",
+                  "name": "<recipient name>",
+                  "type": "to"}],
+    }
+
+See ``config.json`` in the ``community_mailbot`` project repository for a
+working example.
+
+The ``discourse_categories`` script, also in the project, lets you discover
+the category IDs relative to their human-readable names.
+"""
+
+import os
+import json
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+
+import requests
+import mandrill
+
+from community_mailbot.discourse import (CategoryFeed, TopicFeed, TopicCache,
+                                         SiteFeed)
+from community_mailbot.contentpipe import clean_discourse_html
+
+
+def main():
+    args = parse_args()
+
+    with open(args.map_file, 'r') as f:
+        mapping = json.load(f)
+
+    forward_new_topics(mapping, args.cache,
+                       args.url,  args.key, args.user,
+                       args.mandrill, args.cache_only)
+
+
+def parse_args():
+    cache_default = os.getenv(
+        'COMMUNITY_MAILBOT_CACHE',
+        os.path.expandvars('$HOME/.community_mailbot_cache'))
+    default_url = 'http://community.lsst.org'
+    default_mandrill_key = os.getenv('MANDRILL_KEY', None)
+    default_discourse_key = os.getenv('DISCOURSE_KEY', None)
+    default_discourse_user = os.getenv('DISCOURSE_USER', None)
+
+    parser = ArgumentParser(
+        prog='forward_discourse',
+        description=__doc__,
+        formatter_class=RawDescriptionHelpFormatter)
+    parser.add_argument(
+        'map_file',
+        help='Path to JSON file mapping categories to E-mail addresses')
+    parser.add_argument(
+        '--cache',
+        help='Path to the topic cache file ({0})'.format(cache_default),
+        default=cache_default)
+    parser.add_argument(
+        '--key',
+        default=default_discourse_key,
+        help='Discourse API key ({0})'.format(default_discourse_key))
```

Comment:
I guess a good compromise would just be to say that the environment variable can be used in lieu of explicitly setting `--key` etc?